File: Năm 2/Kì 2 đợt 2/ThuatToanATTT/.vscode/Phần 1/bai1.c
#include <stdio.h>
#include <stdlib.h>
int isQPrime(int x){
    int uoc = 2;
    for(int i = 2; i < x; i++){
        if (x%i == 0) uoc++;
        if (uoc > 4) return 0;
    }
    return uoc == 4;
}
int main()
{
    int n;
    printf("Nhap n: ");
    scanf("%d", &n);
    if ( n < 6) printf("Khong co so Q-prime nao!");
    else{
        printf("Cac so Q-prime be hon hoac bang n la: \n");
        int xuongdong = 0;
        for(int i = 6; i <= n; i++ ){
            if(isQPrime(i)){
                xuongdong++;
                printf("%-5d", i);
            }
            if (xuongdong == 5){
                xuongdong = 0;
                printf("\n");
            }
        }
    }
    return 0;
}

File: Năm 2/Kì 2 đợt 2/ThuatToanATTT/.vscode/Phần 1/bai1.py
import math
def isQPrime(q):
    uoc = 2
    for i in range(2, q):
        if q % i == 0: 
            uoc += 1
        if uoc > 4: 
            return 0 
    return uoc == 4

if __name__ == "__main__":
    n = int(input("Nhap N: "))
    if n < 6:
        print("Khong co so Q-prime nao")
    else: 
        count = 0
        for i in range (6, n+1):
            if isQPrime(i) == 1: 
                count += 1
                if count == 1:
                    print("Cac so Q-prime nho hon hoac bang N la")
                print(f"{i:-10d}",end="")
                if count > 0 and count % 5 == 0:
                    print("")
        if count == 0:
            print("Khong co so Q-prime nao nho hon hoac bang N")
